multibranch/src/test/java/org/jenkinsci/plugins/workflow/multibranch/WorkflowMultiBranchProjectTest.java

```diff
@@ -116,4 +125,35 @@
         assertEquals(Collections.<Class<? extends BranchProperty>>emptySet(), clazzes);
     }
 
+    @SuppressWarnings("rawtypes")
+    @Test public void applicableSCMs() throws Exception {
+        final WorkflowMultiBranchProject mp = r.jenkins.createProject(WorkflowMultiBranchProject.class, "p");
+        List<Class> scmTypes = new ArrayList<Class>();
+        List<SCMDescriptor<?>> scmDescriptors = SingleSCMSource.DescriptorImpl.getSCMDescriptors(mp);
+        for (SCMDescriptor<?> scmDescriptor : scmDescriptors) {
+            scmTypes.add(scmDescriptor.clazz);
+        }
+        assertThat(scmTypes, hasItem((Class) GitSCM.class));
+        assertThat(scmTypes, not(hasItem((Class) OldSCM.class)));
+        /* More realistic variant:
```

Comment:
Makes sense to add TODO or something else to explain why we keep this commented code